### ISSUE TITLE
Issue with logging metrics

### ISSUE THREAD

**Issue by zeyaddeeb** [polyaxon/polyaxon#1094]
### Describe the bug
Not able to log metrics or models in custom jobs

```python
You should set an event logger before calling: <function Run.log_metrics at 0x7f99f07dddc0>
Traceback (most recent call last):
  File "model.py", line 89, in <module>
    tracking.log_metrics(**metrics)
  File "/usr/local/lib/python3.8/site-packages/polyaxon/tracking/__init__.py", line 149, in log_metrics
    TRACKING_RUN.log_metrics(step=step, timestamp=timestamp, **metrics)
  File "/usr/local/lib/python3.8/site-packages/polyaxon/client/decorators/no_op.py", line 42, in wrapper
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/polyaxon/client/decorators/offline.py", line 43, in wrapper
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/polyaxon/client/decorators/can_log_event.py", line 45, in wrapper
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/polyaxon/tracking/run.py", line 323, in log_metrics
    self._event_logger.add_events(events)
AttributeError: 'NoneType' object has no attribute 'add_events'
```

### To reproduce

1- Create model.py 

```python
from typing import Any, Dict

import joblib
from polyaxon import tracking
import logging

RANDOM_STATE = 42


logger = logging.getLogger()

def get_metrics() -> Dict[str, Any]:
    return {"accuracy": 1}


if __name__ == "__main__":

    tracking.init(owner="default", project="cool-detection")

    metrics = get_metrics()

    tracking.log_metrics(**metrics)
```

2- Build custom Dockerfile with Polyaxon

```Dockerfile

FROM python:3.8-slim-buster

WORKDIR /app

COPY poetry.lock pyproject.toml ./

RUN poetry config virtualenvs.create false
RUN poetry install

COPY . .


``` 

3- `docker build -t image -f Dockerfile .` && `docker push image`

4- Run `polyaxon run -f polyaxonfile.yaml`

the polyaxonfile

```YAML
version: 1.1
kind: component
tags: [baseline-model]

run:
  kind: job
  container:
    image: xyz.dkr.ecr.us-east-1.amazonaws.com/custom-image:latest
    workingDir: "/app"
    command: ["python", "-u", "model.py"]
```

### Expected behavior
UI will show artifacts and metrics

### Environment
Kubernetes 1.16
Poylaxon 1.1.8

**Comment by mouradmourafiq** [comment]
This is the line that causing you problems:

```python
 tracking.init(owner="default", project="cool-detection")
```

Any reason why you are hard-coding the owner/project values instead of letting Polyaxon inject the correct values based on where the manifest is running.

The reason why this is breaking is that when you pass an owner and project, you are telling Polyaxon to not look for the current information about the run. This could be useful when running inside a notebook where you will create multiple runs programmatically.

I think one sanity check that we should add to the tracking module is to raise an error if a valid run is not correctly initialized, which is the case here, because there's no run_uuid, Polyaxon does not set the correct artifacts path, and that's also why you see the `/tmp` instead of the `/plx-context/artifacts/...`

So to fix your issue you just need to omit the `owner` and `project` from this line `tracking.init(owner="default", project="cool-detection")` and just use `tracking.init()`.

**Comment by zeyaddeeb** [comment]
I tried without hardcoding values as well. The error goes away but nothing appears in logs

**Comment by mouradmourafiq** [comment]
Can you show the logs when you omit those values, either from the UI or using `polyaxon ops logs -a`

**Comment by mouradmourafiq** [comment]
I just used you dockerfile and model.py and it works, I had to change the dockerfile because I don't know about the requirements in the poetry.toml:

```
FROM python:3.8-slim-buster

WORKDIR /app

RUN apt-get update -y && apt-get install -y gcc
Run pip3 install -U polyaxon
Run pip3 install joblib
Run pip3 install numpy
COPY . .
```

model.py

```python
from typing import Any, Dict

import joblib
from polyaxon import tracking
import logging

RANDOM_STATE = 42


logger = logging.getLogger()

def get_metrics() -> Dict[str, Any]:
    return {"accuracy": 1}


if __name__ == "__main__":

    tracking.init()

    metrics = get_metrics()

    tracking.log_metrics(**metrics)
```

<img width="1066" alt="Screenshot 2020-09-24 at 10 48 52" src="https://user-images.githubusercontent.com/1261626/94123034-91cb9380-fe53-11ea-8ddd-ec721a6c4b53.png">

<img width="1437" alt="Screenshot 2020-09-24 at 10 48 11" src="https://user-images.githubusercontent.com/1261626/94123041-942ded80-fe53-11ea-8e15-08e9cf6d3378.png">

**Comment by zeyaddeeb** [comment]
Just container logs, and `tracking.log_metrics(**metrics)` will execute with no output values shown

**Comment by zeyaddeeb** [comment]
> I just used you dockerfile and model.py and it works, I had to change the dockerfile because I don't know about the requirements in the poetry.toml:
> 
> ```
> FROM python:3.8-slim-buster
> 
> WORKDIR /app
> 
> RUN apt-get update -y && apt-get install -y gcc
> Run pip3 install -U polyaxon
> Run pip3 install joblib
> Run pip3 install numpy
> COPY . .
> ```
> 
> model.py
> 
> ```python
> from typing import Any, Dict
> 
> import joblib
> from polyaxon import tracking
> import logging
> 
> RANDOM_STATE = 42
> 
> 
> logger = logging.getLogger()
> 
> def get_metrics() -> Dict[str, Any]:
>     return {"accuracy": 1}
> 
> 
> if __name__ == "__main__":
> 
>     tracking.init()
> 
>     metrics = get_metrics()
> 
>     tracking.log_metrics(**metrics)
> ```
> 
> <img alt="Screenshot 2020-09-24 at 10 48 52" width="1066" src="https://user-images.githubusercontent.com/1261626/94123034-91cb9380-fe53-11ea-8ddd-ec721a6c4b53.png">
> 
> <img alt="Screenshot 2020-09-24 at 10 48 11" width="1437" src="https://user-images.githubusercontent.com/1261626/94123041-942ded80-fe53-11ea-8e15-08e9cf6d3378.png">

Can you share the`polyfile.yml` as well?

**Comment by mouradmourafiq** [comment]
```yaml
version: 1.1
kind: component
tags: [baseline-model]

run:
  kind: job
  container:
    image: mouradmourafiq/foo
    workingDir: "/app"
    command: ["python", "-u", "model.py"]
```
you can run it as is the docker image is public `mouradmourafiq/foo`

**Comment by zeyaddeeb** [comment]
can I set the project in the file or do I have to `polyaxon init project`?

**Comment by mouradmourafiq** [comment]
no you just need to save the `polyaxonfile.yaml` and execute `polyaxon run` if you are in the same folder otherwise `polyaxon run -f path/to/polyaxonfile.yaml`.

If you don't have any project cached you can run with `polyaxon run -p cool-detection [-f path/to/polyaxonfile.yaml]`

**Comment by mouradmourafiq** [comment]
You can also just copy/pasts the yaml content to the UI directly and run it.

**Comment by zeyaddeeb** [comment]
oddly enough that worked.

**Comment by mouradmourafiq** [comment]
I will keep this issue opened and think of enhancement and how to surface issues related to bad configuration to the user instead of using defaults that could cause confusion.

**Comment by polyaxon-team** [comment]
Fixed in #1094
